### ISSUE TITLE
Some issue with editing the colour palatte.

### ISSUE THREAD

**Issue by FlorianJung06** [Zylann/godot_voxel#509]
I am trying to change a colour of the palatte my mesher uses via GD-Script.
Could someone help me out with the line of code?

**Comment by Zylann** [comment]
Can you describe more what you are trying to do? I can only guess...

If you are using `VoxelMesherCubes` and you want to change a specific color of the palette, you have to use the `set_color` function of `VoxelColorPalette`.
```gdscript
mesher.palette.set_color(index, color)
```

**Comment by FlorianJung06** [comment]
Yes, I am using a VoxelMesherCubes and I have setup a palette. However due to the terrain I am trying to produce I'd like to have more then 256 colours alvailable and I wouldn't like to set them up manually.

Okay, so after I attached the generator script to the VoxelTerrains generator I've tried to access the palette in the generator script via this line of code:

self.mesher.palette.set_color(1, Color(1, 0, 1, 1))

All things i have tried haven't worked though. No terrain appears what so ever and there are no erros.

**Comment by Zylann** [comment]
> I'd like to have more then 256 colours alvailable and I wouldn't like to set them up manually.

Palettes with more than 256 colors are not supported. Beyond that amount, you may store colors directly into voxels, using at least 16-bit voxels with the `COLOR_RAW` color mode (which I think is the default), and voxels may store colors encoded using `VoxelTool.color_to_u16`.

> Okay, so after I attached the generator script to the VoxelTerrains generator I've tried to access the palette in the generator script via this line of code: self.mesher.palette.set_color(1, Color(1, 0, 1, 1))

Doing this inside the generator is not valid. The palette is global to everything that gets processed by the mesher, it makes no sense to modify it from inside the generation method of a generator.

> All things i have tried haven't worked though. No terrain appears what so ever and there are no erros.

You are likely not generating voxel values properly. You should set voxels in the `COLOR` channel and encode them.
